HPXMLtoOpenStudio/resources/model.rb

```diff
@@ -941,6 +940,21 @@ def self.merge_unit_models(model, hpxml_osm_map)
         end
       end
     end
+
+    model_objects.each do |obj|
```

Comment:
Currently it feels a bit more complicated than I would have expected just because you're operating on the objects of the final merged model, so you have to know which surfaces are linked and what space to attach them to. I wonder if this can be simplified.